examples/solids/qfunctions/hyperSS.h

```diff
@@ -319,5 +319,95 @@ CEED_QFUNCTION(HyperSSdF)(void *ctx, CeedInt Q, const CeedScalar *const *in,
+
+  // Outputs
+  CeedScalar (*energy) = (CeedScalar(*))out[0];
+  // *INDENT-ON*
+
+  // Context
+  const Physics context = (Physics)ctx;
+  const CeedScalar E  = context->E;
+  const CeedScalar nu = context->nu;
+
+  // Constants
+  const CeedScalar TwoMu = E / (1 + nu);
+  const CeedScalar mu = TwoMu / 2;
+  const CeedScalar Kbulk = E / (3*(1 - 2*nu)); // Bulk Modulus
+  const CeedScalar lambda = (3*Kbulk - TwoMu) / 3;
+
+  // Quadrature Point Loop
+  CeedPragmaSIMD
+  for (CeedInt i=0; i<Q; i++) {
+    // Read spatial derivatives of u
+    // *INDENT-OFF*
+    const CeedScalar du[3][3]   = {{ug[0][0][i],
+                                    ug[1][0][i],
+                                    ug[2][0][i]},
+                                   {ug[0][1][i],
+                                    ug[1][1][i],
+                                    ug[2][1][i]},
+                                   {ug[0][2][i],
+                                    ug[1][2][i],
+                                    ug[2][2][i]}
+                                  };
+    // -- Qdata
+    const CeedScalar wJ         =   qdata[0][i];
+    const CeedScalar dXdx[3][3] = {{qdata[1][i],
+                                    qdata[2][i],
+                                    qdata[3][i]},
+                                   {qdata[4][i],
+                                    qdata[5][i],
+                                    qdata[6][i]},
+                                   {qdata[7][i],
+                                    qdata[8][i],
+                                    qdata[9][i]}
+                                  };
+    // *INDENT-ON*
+
+    // Compute gradu
+    //   dXdx = (dx/dX)^(-1)
+    // Apply dXdx to du = gradu
+    CeedScalar gradu[3][3];
+    for (CeedInt j = 0; j < 3; j++)     // Component
+      for (CeedInt k = 0; k < 3; k++) { // Derivative
+        gradu[j][k] = 0;
+        for (CeedInt m = 0; m < 3; m++)
+          gradu[j][k] += dXdx[m][k] * du[j][m];
+      }
+
+    // Compute Strain : e (epsilon)
+    // e = 1/2 (grad u + (grad u)^T)
+
+    // *INDENT-OFF*
+    const CeedScalar e[3][3] = {{(gradu[0][0] + gradu[0][0])/2.,
+                                 (gradu[0][1] + gradu[1][0])/2.,
+                                 (gradu[0][2] + gradu[2][0])/2.},
+                                {(gradu[1][0] + gradu[0][1])/2.,
+                                 (gradu[1][1] + gradu[1][1])/2.,
+                                 (gradu[1][2] + gradu[2][1])/2.},
+                                {(gradu[2][0] + gradu[0][2])/2.,
+                                 (gradu[2][1] + gradu[1][2])/2.,
+                                 (gradu[2][2] + gradu[2][2])/2.}
+                               };
+    // *INDENT-ON*
+
+    // Strain energy W = 
+    energy[i] = ((lambda/2.+mu)*(e[0][0]*e[0][0] + e[1][1]*e[1][1] + e[2][2]) +
+                 2*mu*(e[0][1]*e[0][1] + e[0][2]*e[0][2] + e[1][2]*e[1][2]))*wJ;
+
+  } // End of Quadrature Point Loop
```

Comment:
This is missing the log term at least.

Reply:
That was a formula mess on my part. I think I repaired it.